src/Parser/Mage/Frost/Modules/Items/ShatteredFragmentsOfSindragosa.js

```diff
@@ -0,0 +1,33 @@
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class ShatteredFragmentsOfSindragosa extends Module {
+
+  static dependencies = {
+		combatants: Combatants,
+	};
+
+  damage = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasHead(ITEMS.SHATTERED_FRAGMENTS_OF_SINDRAGOSA.id);
+  }
+
+  on_byPlayer_damage(event) {
+    if (event.ability.guid !== SPELLS.COMET_STORM_DAMAGE.id) {
+      return;
+    }
```

Comment:
This is trivially wrong if the player has taken Comet Storm (which is very uncommon, I know)

Reply:
Yea, its even more rare for someone to be using Sindragosa and Comet Storm. Im not really sure how to exclude Comet Storm Damage that isnt related to the Legendary considering its the same ID (i checked)

Reply:
I can think of two approaches, I'll write them up when I'm back in front of computer

Reply:
Nevermind, 3 approaches:

1) The easy way out: check if player has Comet Storm, and if so either deactivate this module or replace the stat with "sorry, couldn't calculate because..."

2) Ignore Comet Storm damage that's within 1 second of a Comet Storm cast. This has the chance to clip procs that happen at same time as cast, but it's pretty good. You could even narrow that time window by closely inspecting logs to see the normal delay for a comet storm cast. 

3) Track hard cast count and use the legendaries buff to track proc count. Multiply total comet storm damage by the percentage of procs vs total.

Reply:
I opted for Number 3. Seemed like that was the simplest, best way to get a close approximation of the amount of damage that came from the legendary ... it wont be perfect, but it will be pretty close.